xgi/classes/reportviews.py

```diff
@@ -465,60 +388,57 @@ def __init__(self, hypergraph, bunch=None):
             super().__init__(hypergraph._edge, hypergraph._edge_attr, bunch)
 
     def __call__(self, order):
-        """Filter the results by size."""
+        """Return a new view that keeps track only of the nodes of the given order."""
         return super().__call__(size=order+1)
 
     def members(self, e=None, dtype=list):
-        """Get the nodes that are members of an edge.
-
-        Given an edge ID, this method returns the node IDs
-        that are members of this edge.
+        """Get the node ids that are members of an edge.
 
         Parameters
         ----------
         e : hashable
-            edge ID
-        dtype : list
-            spectify the type of the return value
+            Edge ID.
+        dtype : type, default list
+            Spectify the type of the return value.
 
         Returns
         -------
         list (if dtype is list, default)
-            edge members
+            Edge members.
         dict (if dtype is dict)
-            edge members, if multiple nodes are requested
+            Edge members.
 
         Raises
         ------
-        xgi.XGIError
-            Returns an error if the user tries passing in a slice or if
-            the edge ID does not exist in the hypergraph.
+        XGIError
+            If `e` is a slice or if `e` does not exist in the hypergraph.
+
         """
         if e is None:
             if dtype is dict:
-                return self._id_dict.copy()
+                return {key: self._id_dict[key] for key in self._ids}
             elif dtype is list:
-                return list(self._id_dict.values())
+                return [self._id_dict[key] for key in self._ids]
             else:
                 raise XGIError(f"Unrecognized dtype {dtype}")
 
+        if e not in self._ids:
+            raise XGIError(f"Item {e} not in this view")
+
         try:
             return self._id_dict[e]
-        except KeyError:
-            raise XGIError(f"The edge ID {e} is not in the hypergraph")
-        except TypeError:
+        except KeyError as e:
+            raise XGIError(f"The edge ID {e} is not in the hypergraph") from e
+        except TypeError as e:
             if isinstance(e, slice):
                 raise XGIError(
                     f"{type(self).__name__} does not support slicing, "
                     f"try list(H.edges)[{e.start}:{e.stop}:{e.step}]"
-                )
+                ) from e
 
```

Comment:
Wondering if on line 438 outside the if-statement we should have
```
raise XGIError("The edge ID {e} is not a hashable.") from e
```
and perhaps even remove the prior if statement?